homeassistant/components/velux/__init__.py

```diff
@@ -40,14 +40,19 @@
         )
     return True
 
-
 class VeluxModule:
     """Abstraction for velux component."""
 
     def __init__(self, hass, config):
         """Initialize for velux component."""
+        
+        async def on_hass_stop(event):
+            """Close connection when hass stops."""
+            _LOGGER.info("velux interface terminated")
+            await self.pyvlx.disconnect()                    
+        
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, on_hass_stop)                   
```

Comment:
We try to avoid side effects in init method. Please extract all the side effects into a dedicated setup method.